pkg/apis/acme/v1alpha2/types_issuer.go

```diff
@@ -305,6 +313,25 @@ type ACMEChallengeSolverHTTP01IngressObjectMeta struct {
 	Labels map[string]string `json:"labels,omitempty"`
 }
 
+type ACMEChallengeSolverHTTP01Istio struct {
+	// Optional service type for Kubernetes solver service
+	// +optional
+	ServiceType corev1.ServiceType `json:"serviceType,omitempty"`
+
+	// The names of the gateways that are used to generate the virtualservice
+	// that configures the HTTP01 challenge routes.
+	// `<gateway namespace>/<gateway name>`; specifying a gateway with no
+	// namespace qualifier is the same as specifying the VirtualService's
+	// namespace.
+	// ref: https://github.com/istio/api/blob/24c65c0415b63a6ebca18059c60fc8fccf041e9a/networking/v1beta1/virtual_service.pb.go#L233-L246
+	Gateways []string `json:"gateways,omitempty"`
```

Comment:
Thanks for making this change and for linking to the upstream documentation.
I think the rule is that all slice types should be marked optional: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required



```suggestion
	// The names of the gateways that are used to generate the virtualservice
	// that configures the HTTP01 challenge routes.
	// `<gateway namespace>/<gateway name>`; specifying a gateway with no
	// namespace qualifier is the same as specifying the VirtualService's
	// namespace.
	// ref: https://github.com/istio/api/blob/24c65c0415b63a6ebca18059c60fc8fccf041e9a/networking/v1beta1/virtual_service.pb.go#L233-L246
	// +optional
	Gateways []string `json:"gateways,omitempty"`
```


And then I wondered if it should perhaps be a required field?
Certainly it should if the the Istio CRD / Webhooks implement any validation enforcing that `VirtualService.Spec.Gateways` is required, but I don't see any validation, so perhaps not.

What would happen if a user left this value empty? Is there some Default Gateway setitng in Istio?

Reply:
"When this field is omitted, the default gateway (mesh) will be used, which would apply the rule to all sidecars in the mesh."
https://istio.io/latest/docs/reference/config/networking/virtual-service/#VirtualService

So, it is perfectly valid to not provide any gateways here.